ci-operator/step-registry/cucushift/installer/check/ibmcloud/kms-key/cucushift-installer-check-ibmcloud-kms-key-commands.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    rg=$1
+    export region
+    "${IBMCLOUD_CLI}" config --check-version=false
+    echo "Try to login to ${rg}..."
+    "${IBMCLOUD_CLI}" login -r ${region} -g ${rg} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+RESOURCE_GROUP=$(jq -r .ibmcloud.resourceGroupName ${SHARED_DIR}/metadata.json)
+
+ibmcloud_login ${RESOURCE_GROUP}
+
+critical_check_result=0
+
+key_file="${SHARED_DIR}/ibmcloud_key.json"
+
+id=$(jq -r .id ${key_file})
+if [[ -z $id ]]; then
+    echo "[ERROR] fail to find kp instance id !!"
+    exit 1
+fi
+
+#check the kpKey whether used in the volumes of the nodes(master & worker)
+mapfile -t vols < <(ibmcloud kp registrations -i ${id} -o JSON  | jq -r .[].resourceCrn)
+echo "INFO: key registrations list is: ${#vols[@]}" "${vols[@]}"
+
+#check that node os disk is encrypted
+machines=$(oc get machines -A --no-headers | awk '{print $2}')
+if [[ $(echo "${machines}" | wc -l) -ne 6 ]]; then
```

Comment:
I do not think we can always assume the cluster has 6 machines, in future, maybe we have compact, sno, or a customized worker replicas.

Reply:
-gt 0 ?

Reply:
yes